pkg/webhooks/machine_webhook.go

```diff
@@ -1381,6 +1381,16 @@ func validateAzureDataDisks(machineName string, spec *machinev1.AzureMachineProv
 			errs = append(errs, field.Invalid(fldPath.Child("lun"), disk.Lun, "each Data Disk must have a unique lun"))
 		}
 
+		switch disk.DeletionPolicy {
+		case machinev1.DiskDeletionPolicyTypeDelete, machinev1.DiskDeletionPolicyTypeDetach:
+			// Valid scenarios, do nothing
+		case "":
+			errs = append(errs, field.Required(fldPath.Child("deletionPolicy"), "deletionPolicy must be provided"))
```

Comment:
Should probably add the valid values in here to help the end user

Reply:
Yeah, I followed what was the pattern in the other validations. But I'm happy to add that.
Added.